src/firefox/preferences.js

```diff
@@ -117,3 +117,33 @@ export function getPrefs(
     ...appPrefs,
   };
 }
+
+// Yargs coerce function for run command custom-prefs
+export function coerceCLICustomPreference(
```

Comment:
To make flow happy about the type checking of this function, it needs a small rewrite:
```
export function coerceCLICustomPreference(
  cliPrefs: string | Array<string>
): FirefoxPreferences {
  let customPrefs = {};

  if (!Array.isArray(cliPrefs)) {
    cliPrefs = [cliPrefs];
  }

  for (let pref of cliPrefs) {
    let [key, value] = pref.split('=');
    // Check the property key.
    if (/[^\w{@}.-]/.test(key)) {
      throw new UsageError(`Invalid custom preference name: ${key}`);
    }

    if (value === `${parseInt(value)}`) {
      value = parseInt(value, 10);
    } else if (value === 'true' || value === 'false') {
      value = (value === 'true');
    }

    customPrefs[`${key}`] = value;
  }

  return customPrefs;
}
```

The function esplicitly return a `FirefoxPreferences` type, and flow is able using its type inference capability that the `for...of` loop produce an object and its properties have the expected value types (boolean, number or string), which is the definition of the type FirefoxPreferences.